sound/soc/sof/intel/cnl.c

```diff
@@ -230,6 +253,8 @@ struct snd_sof_dsp_ops sof_cnl_ops = {
 	.pcm_hw_params	= hda_dsp_pcm_hw_params,
 	.pcm_trigger	= hda_dsp_pcm_trigger,
 
+	.dai_fixup	= cnl_dai_fixup,
+
```

Comment:
does this belong in the SOF ops, or could you use the existing ASoC .fixup?

Reply:
I check the fixup in ASoc. It is not fit us. for three reasons
(1) the input params are not fit for link_hda_load
(2) the link dma channel is not returned by these functions
(3) these functions are called at the time of hw_param

As you know, hda config is set at core layer and hda dma channel info could be only gotten at intel layer. It is a trouble I get, so I add a ops to get hda info from Intel layer. And I think other vendors may also want a ops to get platform info if they use SOF.

If it is not good, how about my last RFC about this, It is done in hw_params. If it is a candidate, I will refine it later.

Reply:
Sorry @RanderWang , I don't get your answers.
1) If you look at the SST and machine drivers, they've used the tx- and rx-masks for stream tags, so why are the parameters for fixup not right? It may not be elegant to abuse a definition but it's been done before.
2) why would the DMA channel be returned by the fixup when you want to pass it to firmware using the fixup?
3) to paraphrase your answer, the functions are called during hw_params, and your last counter proposal is to say "how about my last RFC about this, It is done in hw_params". there is a serious logic issue here that I can't sort out without clarifications.

Reply:
@plbossart  
(1) ok, I get your idea.  
(2) The function is called dai_config, and was changed to fixup.
(3) please ignore my question